spyder/plugins/projects/utils/watcher.py

```diff
@@ -23,6 +24,30 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseThreadWrapper(watchdog.utils.BaseThread):
+    """
+    Wrapper around watchdog BaseThread class.
+
+    """
+    queue = None
+
+    def __init__(self):
+        super(BaseThreadWrapper, self).__init__()
+        self._original_run = self.run
+        self.run = self.run_wrapper
+
+    def run_wrapper(self):
+        try:
+            self._original_run()
+        except OSError as e:
+            logger.exception('Watchdog thread exited with error %s',
+                             e.strerror)
+            self.queue.put(e)
+
+
```

Comment:
Please add this comment here:

    # Monkeypatching BaseThread to prevent the error reported in
    # spyder-ide/spyder#11370